code/modules/mob/living/carbon/xenomorph/xenoprocs.dm

```diff
@@ -580,6 +580,27 @@
 	if(pipe)
 		handle_ventcrawl(pipe)
 
+// Hunter Vent Crawl
+/mob/living/carbon/xenomorph/hunter/vent_crawl()
+	set name = "Crawl through Vent"
+	set desc = "Enter an air vent and crawl through the pipe system."
+	set category = "Alien"
+	if(!check_state())
+		return
+	var/pipe = start_ventcrawl()
+	if(pipe)
+		handle_ventcrawl(pipe, HUNTER_VENT_CRAWL_TIME, TRUE)
+
+// Larva Vent Crawl
+/mob/living/carbon/xenomorph/larva/vent_crawl()
+	set name = "Crawl through Vent"
+	set desc = "Enter an air vent and crawl through the pipe system."
+	set category = "Alien"
+	if(!check_state())
+		return
+	var/pipe = start_ventcrawl()
+	if(pipe)
+		handle_ventcrawl(pipe, LARVA_VENT_CRAWL_TIME, TRUE)
```

Comment:
maybe make vent crawl a var on xeno caste? 

this is a lot of duplicate code

Reply:
Not a bad idea, though there's already a headache inducing amount of vars which I hate adding to, but sure.